libraries/standard/mqtt/utest/mqtt_lightweight_utest.c

```diff
@@ -1804,6 +1897,31 @@ void test_MQTT_SerializePublishHeader( void )
     /* Payload should not be serialized. */
     TEST_ASSERT_EQUAL_MEMORY( expectedPacket, &buffer[ BUFFER_PADDING_LENGTH ], packetSize );
     checkBufferOverflow( buffer, sizeof( buffer ) );
+
+
+    /* Again with QoS2 and dup. */
+    publishInfo.qos = MQTTQoS2;
+    publishInfo.dup = true;
+    status = MQTT_GetPublishPacketSize( &publishInfo, &remainingLength, &packetSize );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+    padAndResetBuffer( buffer, sizeof( buffer ) );
+    status = MQTT_SerializePublishHeader( &publishInfo,
+                                          PACKET_ID,
+                                          remainingLength,
+                                          &fixedBuffer,
+                                          &headerSize );
+    TEST_ASSERT_EQUAL_INT( MQTTSuccess, status );
+
+    memset( ( void * ) expectedPacket, 0x00, sizeof( expectedPacket ) );
+    pIterator = expectedPacket;
+    /* Dup = 0x8, QoS2 = 0x4, 8 + 4 = 0xC. */
```

Comment:
Are these lengths? If so, would you please mention so.

Reply:
No they are flags.